wordpress-mobile/WordPress-iOS
# iPad Splitview: Me and My Sites

Introduces `UISplitViewController` to the Me and My Sites sections.

As we've discussed previously, this tries to mimic the system Settings app and Calypso as much as possible. So the left pane always shows the top level navigation, and the right pane can be used to navigate into each item. The left pane is currently always visible in both portrait and landscape, as I figured we have enough space to do so.

I've made a number of customisations to `UISplitViewController` to support the behaviour we require, in the subclass `WPSplitViewController`. For example, it handles wrapping and unwrapping child view controllers in navigation controllers where required. It also handles automatically populating the right pane with a detail view controller when necessary, through a protocol: `WPSplitViewControllerDetailProvider`. A primary view controller (left pane) can implement the method

```
func initialDetailViewControllerForSplitView(splitView: WPSplitViewController) -> UIViewController?`
```

And return a view controller that should populate the right pane on first appearance. For example, the Me section shows the My Profile detail screen, and a blog details VC shows the Stats section.

I'm currently not firing analytics when automatically populating the detail pane, as I figure the user hasn't explicitly requested to see these screens. This may be an incorrect assumption though, so let me know if we should change that.

Changing to UISplitViewController touches a lot of the app's initialisation after login, so we need to ensure that nothing has changed / broken there. It also affects iPhone and iPad equally, as we use UISplitViewController on both devices (it's just collapsed on iPhone).

### To test:

* Try the Me and My Sites sections out on both iPhone and iPad, moving between orientations and multitasking configurations.
* Try to follow every navigation route within each section to ensure things transition as you'd expect.
* My Sites is more complex than Me, in that it has two-level primary navigation. Like Calypso, you can switch between the blogs list and blog detail in the left pane, and view sections like Stats and Posts in the right hand pane.
* Check logging into a fresh install, logging out and back in, and check that things initialise as you'd expect.
* Check the shortcuts that jump straight to specific sections – these are in the methods in `WPTabBarController` that begin `switchTo...`. They can be triggered by:
  * Writing a new post and then saving / posting it. You should get taken to the post list for that blog.
  * 3D touching on the app icon on the home screen and choosing Stats.
  * The Theme shortcuts aren't actually used at the moment, as they were part of Plans post-purchase.
* Test on iPhone 6s+ too, as it shows the split in landscape by default.

Needs review: @kurzee
